command/agent/config.go

```diff
@@ -1371,6 +1376,9 @@ func MergeConfig(a, b *Config) *Config {
 	if b.NodeID != "" {
 		result.NodeID = b.NodeID
 	}
+	if b.DisableHostNodeID == true {
+		result.DisableHostNodeID = b.DisableHostNodeID
```

Comment:
It's close - probably needs some support for a few other data types that show up in the config options.